src/main/grammars/ElmParser.bnf

```diff
@@ -167,12 +167,21 @@ private InternalValueDeclarationLeft ::=
     | Pattern
 
 FunctionDeclarationLeft ::=
-    LOWER_CASE_IDENTIFIER Pattern*
+    LOWER_CASE_IDENTIFIER FunctionDeclarationPattern*
 
  // TODO [drop 0.18] remove OperatorDeclarationLeft
 OperatorDeclarationLeft ::=
-    OperatorAsFunction_inner Pattern*
-
+    OperatorAsFunction_inner FunctionDeclarationPattern*
+
+private FunctionDeclarationPattern ::=
+        AnythingPattern
+        | LowerPattern
+        | TuplePattern
+        | Unit
+        | ListPattern // Always a partial pattern, but not a syntax error
```

Comment:
Since elm doesn't allow multiple function declarations as pattern cases like Haskell does, there's no way to use numeric or list patterns in function parameters. The 0.18 compiler used the term "partial pattern" for these kinds of error. Now it looks like this:

```
This pattern does not cover all possiblities:

16| foo [] = ()
        ^^
Other possibilities include:

    _ :: _
```

or

```
This pattern does not cover all possiblities:

16| foo [x] = ()
        ^^^
Other possibilities include:

    []
```